tendrl/commons/objects/node_context/__init__.py

```diff
@@ -120,31 +120,23 @@ def on_change(self, attr, prev_value, current_value):
                 )
                 # Load cluster_node_context will load node_context
                 # and it will be updated with latest values
-                cluster_node_context = NS.tendrl.objects.ClusterNodeContext(
-                    node_id=self.node_id,
-                    integration_id=_tc.integration_id
-                )
-                cluster_node_context.save()
-                del cluster_node_context
-                global_details = NS.tendrl.objects.GlobalDetails(
-                    integration_id=_tc.integration_id).load()
-                if global_details.status.lower() == "healthy":
-                    global_details.status = "unhealthy"
-                    global_details.save()
-                    _cluster = NS.tendrl.objects.Cluster(
+                _cnc_old = \
+                    NS.tendrl.objects.ClusterNodeContext(
+                        node_id=self.node_id,
                         integration_id=_tc.integration_id
                     ).load()
-                    msg = "Cluster:%s is %s" % (
-                        _cluster.short_name, "unhealthy")
-                    instance = "cluster_%s" % _tc.integration_id
-                    event_utils.emit_event(
-                        "cluster_health_status",
-                        "unhealthy",
-                        msg,
-                        instance,
-                        'WARNING',
-                        integration_id=_tc.integration_id
+                _cnc_new = \
+                    NS.tendrl.objects.ClusterNodeContext(
+                        node_id=self.node_id,
+                        integration_id=_tc.integration_id,
+                        first_sync_done=_cnc_old.first_sync_done,
```

Comment:
Why these fields `first_sync_done` and `is_managed` here?

Reply:
inside ClusterNodeContext we are actually loading node_context_obj and populating the values into ClusterNodeContext, but first_sync_done and is_managed values are not come from node_context for i am loading ClusterNodeContext and  while creating _cnc_new it will populate other fields from node_context and these two i am passing from loaded _cnc_old.

if i save directly without passing these two then other fields have a value from node_context but these two fields will come None.

Reply:
ok. this is a workaround as such till we keep populating ClusterNodeContext from NodeContext